tests/test_interface.rb

```diff
@@ -556,28 +555,26 @@ def test_interface_duplex_change
     interface.duplex = 'auto'
     assert_equal('auto', interface.duplex)
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
   end
 
   def test_interface_duplex_invalid
-    skip('Duplex changes not supported on IOS XR') if platform == :ios_xr
     interface = Interface.new(interfaces[0])
     interface.speed = 1000
     assert_raises(RuntimeError) { interface.duplex = 'hello' }
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
   end
 
   def test_interface_duplex_valid
-    skip('Duplex changes not supported on IOS XR') if platform == :ios_xr
     interface = Interface.new(interfaces[0])
     interface.speed = 1000
     interface.duplex = 'full'
     assert_equal('full', interface.duplex)
     interface_ethernet_default(interfaces_id[0])
-  rescue RuntimeError => e
+  rescue Cisco::CliError => e
     speed_change_disallowed(e.message)
```

Comment:
Yup - I merged in the latest develop content prior to these update diffs. The reason for this change is that RuntimeError includes both CliError and UnsupportedError so we were passing the UnsupportedError through to speed_change_disallowed and failing the test. Catching only CliError gives us the more correct behavior.